hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -233,6 +233,10 @@ protected Dialect() {
 		registerHibernateType( Types.CLOB, StandardBasicTypes.CLOB.getName() );
 		registerHibernateType( Types.REAL, StandardBasicTypes.FLOAT.getName() );
 
+		if(supportsPartitionBy()) {
+			registerKeyword( "PARTITION" );
+		}
+
```

Comment:
what about adding directly  the ` registerKeyword( "PARTITION" );` to the dialects supporting it the